Azure.MgmtSdk.Analyzers.Test/DataPropertyEtagAnalyzerTests.cs

```diff
@@ -0,0 +1,57 @@
+﻿using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System.Threading.Tasks;
+using VerifyCS = AzureMgmtSDKAnalyzer.Test.CSharpAnalyzerVerifier<
+    Azure.MgmtSdk.Analyzers.DataPropertyEtagAnalyzer>;
+
+namespace Azure.MgmtSdk.Analyzers.Test
+{
+    [TestClass]
+    public class DataPropertyEtagTests
+    {
+        [TestMethod]
+        public async Task AZM0043DataPropertyEtagCorrectVariableType()
+        {
+            var test = @"namespace Azure.ResourceManager.Network
+{
+    public partial class Test
+    {
+        public class ETag
+        {
+        }
+#nullable enable
+        public ETag? Etag { get; }
+#nullable disable
+    }
+}";
+            await VerifyCS.VerifyAnalyzerAsync(test); // Default No errors.
+        }
+
+        [TestMethod]
+        public async Task AZM0043DataPropertyEtagInCorrectPropertyType()
+        {
+            var test = @"namespace Azure.ResourceManager.Network
+{
+    public partial class Test
+    {
+        public string Etag { get; set; }
```

Comment:
I think we actually cannot check a value of a property in static code analysis, because it requires real testing. Below is the rule:

• The data type of a property should be 'ETag' if its name is 'etag' and **the value is really a ETag**. The property name  should be 'ETag'(Optional). [See Sample](https://apiview.dev/Assemblies/Review/25a98e223bda4f39b4ef76bebffa2c23#c33ff8a1778c4021b8dbc70652445d97)

I think the rule means if a property name is `etag` and the returned value in reality is a ETag value (e.g., `efe5421efsac`), then we should check if its type is `ETag`. As you can see, during code analysis, we don't know that. It requires manual testing.